### PR TITLE
Add Test Coverage tab for Communication

### PR DESCRIPTION
**Pull request by minnieliu**

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -159,10 +159,34 @@ jobs:
 
       - template: ../steps/upload-repository-on-failure.yml
 
+      - task: Maven@3
```
Comment (srnagar) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:162]: nit: Move this task to run immediately after the `'Run tests: $(DisplayName)'` task is complete.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 2]
```diff
@@ -159,10 +159,34 @@ jobs:
...
         inputs:
           mergeTestResults: true
           testRunTitle: 'Live tests for ${{ parameters.ServiceDirectory }} $(DisplayName)'
           ${{ if ne(parameters.TestResultsFiles, '') }}:
             testResultsFiles: ${{ parameters.TestResultsFiles }}
+
+      # Azure DevOps only seems to respect the last code coverage result published, so only do this for Windows + Java LTS.
+      # Code coverage reporting is setup only for Track 2 modules.
+      - task: PublishCodeCoverageResults@1
+        condition: and(eq(variables['OsName'], 'Windows'), eq(variables['JavaTestVersion'], '1.11'))
```
Comment (srnagar) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:187]: Add this condition here as well - `, eq('${{ parameters.ServiceDirectory }}', 'communication')`

**Review by srnagar — APPROVED** [review]
A couple of minor updates required. Otherwise, looks good.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml (hunk 1 above)
Comment (minnieliu) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:152]: Done, thanks Srikanta

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml (hunk 2 above)
Comment (minnieliu) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml]: Done, thanks Srikanta

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 3]
```diff
@@ -149,6 +149,20 @@ jobs:
           ${{ insert }}: ${{ parameters.TestStepMavenInputs }}
         env: ${{ parameters.EnvVars }}
 
+      - task: Maven@3
+        displayName: 'Generate aggregate code coverage report'
+        condition: and(eq(variables['OsName'], 'Windows'), eq(variables['JavaTestVersion'], '1.11'), eq('${{ parameters.ServiceDirectory }}', 'communication'))
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:154]: We should avoid hardcoding checks specific to one service in our common templates. Instead we should have an opt-in or opt-out that can be set in the ci.yml for a given service.

**Pull request merged**